app/controllers/blogs_controller.rb

```diff
@@ -3,13 +3,19 @@
 class BlogsController < ApplicationController
   skip_before_action :authenticate_user!, only: %i[index show]
 
-  before_action :set_blog, only: %i[show edit update destroy]
+  before_action :set_current_user_blog, only: %i[edit update destroy]
 
   def index
     @blogs = Blog.search(params[:term]).published.default_order
   end
 
-  def show; end
+  def show
+    @blog = if current_user
+              Blog.where(id: params[:id]).where("user_id = #{current_user.id} OR secret = false").take!
```

Comment:
`user_id = #{current_user.id}` みたいな書き方はSQLインジェクションを招きかねないので、使わないようにしてください。
（というのがこのプラクティスの学習テーマですよ！！）

あとtake!を使わなくても

```suggestion
              Blog.where("user_id = #{current_user.id} OR secret = false").find(params[:id])
```

のように最後にfindすれば同じ結果になるはずです

Reply:
- 書き方を修正しました。（学習テーマの件、スミマセン😓）
- 下記の「scopeを使う書き方に修正」を行い、大部分を`models/brog.rb`に移動しました。